docs/en/creating-widgets/supplemental.md

```diff
@@ -0,0 +1,891 @@
+
+## Handling Focus
+
+When outputting `VNode`s, widgets can use `VNodeProperties`'s `focus` property to control whether the resulting DOM element should receive focus when rendering. This is a special property that accepts either a `boolean` or a function that returns a `boolean`.
+
+When passing `true` directly, the element will only receive focus when the previous value was something other than `true` (similar to [regular property change detection](#changing-properties-and-diff-detection)). When passing a function, the element will receive focus when `true` is returned, regardless of what the previous return value was.
+
+For example:
+
+Given element ordering, the following 'firstFocus' input will receive focus on the initial render, whereas the 'subsequentFocus' input will receive focus for all future renders as it uses a function for its `focus` property.
+
+> src/widgets/FocusExample.tsx
+
+**Function-based variant:**
+
+```tsx
+import { create, tsx, invalidator } from '@dojo/framework/core/vdom';
+
+const factory = create({ invalidator });
+
+export default factory(function FocusExample({ middleware: { invalidator } }) {
+	return (
+		<div>
+			<input key="subsequentFocus" type="text" focus={() => true} />
+			<input key="firstFocus" type="text" focus={true} />
+			<button onclick={() => invalidator()}>Re-render</button>
+		</div>
+	);
+});
+```
+
+**Class-based variant:**
+
+```tsx
+import WidgetBase from '@dojo/framework/core/WidgetBase';
+import { tsx } from '@dojo/framework/core/vdom';
+
+export default class FocusExample extends WidgetBase {
+	protected render() {
+		return (
+			<div>
+				<input key="subsequentFocus" type="text" focus={() => true} />
+				<input key="firstFocus" type="text" focus={true} />
+				<button onclick={() => this.invalidate()}>Re-render</button>
+			</div>
+		);
+	}
+}
+```
+
+### Delegating focus
+
+Function-based widgets can use the [`focus` middleware](../middleware/supplemental.md#focus) to provide `focus` to their children or to accept `focus` from a parent widget.
+
+Class-based widgets can use the `FocusMixin` (from `@dojo/framework/core/mixins/Focus`) to delegate focus in a similar way.
+
+The `FocusMixin` adds `focus` and `shouldFocus` to a widget's API. `shouldFocus` checks if the widget is in a state to perform a focus action and will only return `true` once, until the widget's `focus` method has been called again. This `shouldFocus` method is designed to be passed to child widgets or nodes as the value of their `focus` property.
+
+When `shouldFocus` is passed to a widget, it will be called as the properties are set on the child widget, meaning that any other usages of the parent's `shouldFocus` method will result in a return value of `false`.
+
+An example usage controlling focus across child VNodes (DOM) and WNodes (widgets):
+
+> src/widgets/FocusableWidget.tsx
+
+```tsx
```

Comment:
Do we need a function-based variant example here?

Reply:
I think that would be in the middleware guide.